app/models/generic_object.rb

```diff
@@ -15,62 +13,43 @@ def initialize(attributes = {})
     super
   end
 
-  def inspect
-    attributes_as_string = self.class.column_names.collect do |name|
-      "#{name}: #{attribute_for_inspect(name)}"
-    end
-
-    attributes_as_string += custom_attributes.collect do |ca|
-      "#{ca.name}: #{ca_attribute_for_inspect(custom_attribute_getter(ca.name))}"
-    end
-
-    "#<#{self.class} #{attributes_as_string.join(", ")}>"
-  end
-
-  def property_attributes
-    custom_attributes.each_with_object({}) { |ca, h| h[ca.name] = custom_attribute_getter(ca.name) }
-  end
-
   def property_attributes=(options)
     raise "generic_object_definition is nil" unless generic_object_definition
     options.keys.each do |k|
       unless property_attribute_defined?(k)
         raise ActiveModel::UnknownAttributeError.new(self, k)
       end
     end
-    options.each { |k, v| custom_attribute_setter(k.to_s, v) }
+    super
+  end
+
+  def inspect
+    attributes_as_string = self.class.column_names.collect do |name|
+      "#{name}: #{attribute_for_inspect(name)}"
+    end
+
+    "#<#{self.class} #{attributes_as_string.join(", ")}>"
```

Comment:
If we're rewriting inspect, we may want to pull out the object_id... See `Object.new.inspect`.  One trick to getting that at the ActiveRecord level is to do the following:

```ruby
def inspect
  # ...

  prefix = Kernel.instance_method(:inspect).bind(VmThing.new).call.split(' ', 2).first
  "#{prefix} #{attributes_as_string.join(", ")}>"
end